src/detect-dce-stub-data.c

```diff
@@ -167,6 +167,9 @@ void DetectDceStubDataRegister(void)
 
 static int DetectDceStubDataSetup(DetectEngineCtx *de_ctx, Signature *s, const char *arg)
 {
+    if (DetectSignatureSetAppProto(s, ALPROTO_DCERPC) < 0)
```

Comment:
This will fail to load a valid rule like `alert smb any any -> any any (flow:to_client; dce_stub_data; content:"|00 00 02 00|"; sid:1;)`

SMB and DCERPC are an special case, as we can have ALPROTO_SMB and ALPROTO_DCERPC.

Reply:
oh, thanks for noticing. Extending the check